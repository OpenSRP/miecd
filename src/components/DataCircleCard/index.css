.compartment-wrapper .circlesRow .circle {
    list-style-type: none;
    width: 7.5rem;
    height: 7.5rem;
    border-radius: 3.75rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.compartment-wrapper .circlesRow .circle:hover {
    cursor: pointer;
}

.compartment-wrapper .card-body > p {
    margin-top: 12%;
}

.compartment-wrapper .dataCircleCard {
    padding: 0;
    background-color: #ffffff;
    border: 0.063em solid #dadce0;
    border-radius: 0.313rem;
    box-shadow: 0 0.063em 0.188em 0 #d7d7e0;
    height: 14.375rem;
}

.compartment-wrapper .circlesRow {
    display: flex;
    justify-content: space-around;
    align-items: center;
}

.compartment-wrapper .card-group > .card {
    flex: 1 1 auto;
}

.compartment-wrapper .card {
    border: 0px;
    margin: 20px 50px 20px 0;
}

.compartment-wrapper .circlesRow > .red {
    background-color: #ff1e1e;
}

.compartment-wrapper .circlesRow > .yellow {
    background-color: #ffd300;
}

.compartment-wrapper .circlesRow > .green {
    background-color: #37ae12;
}

.compartment-wrapper .circlesRow > .total-children {
    background-color: #637381;
}

.compartment-wrapper .circlesRow > .stunting {
    background-color: #ff6361;
}

.compartment-wrapper .circlesRow > .wasting {
    background-color: #003f5c;
}

.compartment-wrapper .circlesRow > .overweight {
    background-color: #58508d;
}

.compartment-wrapper .circlesRow > .inappropriate-feeding {
    background-color: #ceb01cd2;
}

.compartment-wrapper .circlesRow > .normal {
    background-color: #38736e;
}

.compartment-wrapper .circlesRow > .green,
.compartment-wrapper .circlesRow > .red,
.compartment-wrapper .circlesRow > .orange {
    border: none !important;
}

.compartments .card-title {
    border-bottom: 0.063em solid #dadce0;
    color: #001429;
    font-size: 1.125rem;
    font-weight: 400;
    text-align: left;
    padding: 1rem;
}

.compartment-wrapper span.number {
    color: #ffffff;
    font-size: 1.3rem;
    font-weight: 400;
}

.compartment-wrapper span.risk-level {
    color: #ffffff;
    font-size: 1rem;
    font-weight: 400;
}

.compartment-wrapper .circleRow > a,
.compartment-wrapper a:hover {
    color: inherit;
    text-decoration: inherit;
}

.card_title {
    font-size: 1.3rem;
    text-align: center;
}
