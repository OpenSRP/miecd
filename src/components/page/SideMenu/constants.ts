import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { NavigationModule, PageLink } from './SubMenu';
import { TFunction } from 'react-i18next';
import {
    ANC_URL,
    CHILD_URL,
    CLIENT_URL,
    HOME_URL,
    HOUSEHOLD_URL,
    LOCATIONS_URL,
    NBC_AND_PNC_ANALYSIS_URL,
    NBC_AND_PNC_COMPARTMENTS_URL,
    NBC_AND_PNC_LOGFACE_URL,
    NBC_AND_PNC_URL,
    NUTRITION_ANALYSIS_URL,
    NUTRITION_COMPARTMENTS_URL,
    NUTRITION_LOGFACE_URL,
    NUTRITION_URL,
    PREGNANCY_ANALYSIS_URL,
    PREGNANCY_COMPARTMENTS_URL,
    PREGNANCY_LOGFACE_URL,
    PREGNANCY_URL,
    REPORTS_URL,
    ROLE_URL,
    TEAM_URL,
    USER_URL,
} from '../../../constants';

import {
    ENABLE_ANC_PAGE,
    ENABLE_CHILD_PAGE,
    ENABLE_CLIENT_PAGE,
    ENABLE_HOUSEHOLD_PAGE,
    ENABLE_LOCATION_PAGE,
    ENABLE_ROLE_PAGE,
    ENABLE_TEAM_PAGE,
    ENABLE_USER_PAGE,
} from '../../../configs/env';
import { Dictionary } from '@onaio/utils';

export const navigationPageLinks = (t: TFunction): Dictionary<PageLink> => {
    return {
        // Page links
        CLIENT_PAGE_NAVIGATION: {
            label: t('All clients'),
            url: CLIENT_URL,
        },
        HOUSEHOLD_PAGE_NAVIGATION: {
            label: t('Household'),
            url: HOUSEHOLD_URL,
        },
        ANC_PAGE_NAVIGATION: {
            label: t('ANC'),
            url: ANC_URL,
        },
        CHILD_PAGE_NAVIGATION: {
            label: t('Child'),
            url: CHILD_URL,
        },

        USERS_PAGE_NAVIGATION: {
            label: t('Users'),
            url: USER_URL,
        },
        ROLES_PAGE_NAVIGATION: {
            label: t('Roles'),
            url: ROLE_URL,
        },
        TEAMS_PAGE_NAVIGATION: {
            label: t('Teams'),
            url: TEAM_URL,
        },
        LOCATIONS_PAGE_NAVIGATION: {
            label: t('Locations'),
            url: LOCATIONS_URL,
        },

        PREGNANCY_PAGE_NAVIGATION: {
            label: t('Pregnancy'),
            url: PREGNANCY_URL,
        },

        NUTRITION_PAGE_NAVIGATION: {
            label: t('Nutrition'),
            url: NUTRITION_URL,
        },

        NEWBORN_AND_POSTNATAL_PAGE_NAVIGATION: {
            label: t('NBC & PNC'),
            url: NBC_AND_PNC_URL,
        },

        PREGNANCY_COMPARTMENTS_PAGE_NAVIGATION: {
            label: t('Compartments'),
            url: PREGNANCY_COMPARTMENTS_URL,
        },

        NBC_AND_PNC_COMPARTMENTS_PAGE_NAVIGATION: {
            label: t('Compartments'),
            url: NBC_AND_PNC_COMPARTMENTS_URL,
        },

        NUTRITION_COMPARTMENTS_PAGE_NAVIGATION: {
            label: t('Compartments'),
            url: NUTRITION_COMPARTMENTS_URL,
        },

        NBC_AND_PNC_LOGFACE_PAGE_NAVIGATION: {
            label: t('Log face'),
            url: NBC_AND_PNC_LOGFACE_URL,
        },

        PREGNANCY_LOGFACE_PAGE_NAVIGATION: {
            label: t('Log face'),
            url: PREGNANCY_LOGFACE_URL,
        },

        NUTRITION_LOGFACE_PAGE_NAVIGATION: {
            label: t('Log face'),
            url: NUTRITION_LOGFACE_URL,
        },

        PREGNANCY_ANALYSIS_PAGE_NAVIGATION: {
            label: t('Analysis'),
            url: PREGNANCY_ANALYSIS_URL,
        },

        NBC_AND_PNC_ANALYSIS_PAGE_NAVIGATION: {
            label: t('Analysis'),
            url: NBC_AND_PNC_ANALYSIS_URL,
        },

        NUTRITION_ANALYSIS_PAGE_NAVIGATION: {
            label: t('Analysis'),
            url: NUTRITION_ANALYSIS_URL,
        },
    };
};

// icons
export const homeNavIcon: IconProp = ['fas', 'home'];
export const pregancyNavIcon: IconProp = ['far', 'user'];
export const nbcandpncNavIcon: IconProp = ['far', 'user'];
export const nutritionNavIcon: IconProp = ['far', 'user'];
export const newbornAndPostnatalNavIcon: IconProp = ['far', 'user'];
export const clientModuleNavIcon: IconProp = ['far', 'user'];
export const reportModuleNavIcon: IconProp = ['fas', 'chart-line'];
export const adminModuleNavIcon: IconProp = ['fas', 'cog'];

// Navigation links for navigation module. collapses when clicked to display the pageLinks
export const modulePageLinks = (t: TFunction) => {
    const links = {
        HOME_PARENT_NAV: {
            icon: homeNavIcon,
            label: t('Home'),
            url: HOME_URL,
        },
        NBC_AND_PNC_PARENT_NAV: {
            icon: nbcandpncNavIcon,
            label: t('NBC & PNC'),
            url: NBC_AND_PNC_URL,
        },
        NUTRITION_PARENT_NAVIGATION: {
            icon: nutritionNavIcon,
            label: t('Nutrition'),
            url: NUTRITION_URL,
        },
        PREGNANCY_MODULE_PARENT_NAV: {
            icon: pregancyNavIcon,
            label: t('Pregnancy'),
            url: PREGNANCY_URL,
        },
        CLIENT_MODULE_PARENT_NAV: {
            icon: clientModuleNavIcon,
            label: t('Client Records'),
        },
        REPORT_MODULE_PARENT_NAV: {
            icon: reportModuleNavIcon,
            label: t('Reports'),
            url: REPORTS_URL,
        },
        ADMIN_MODULE_PARENT_NAV: {
            icon: adminModuleNavIcon,
            label: t('Admin'),
        },
    };
    return links;
};

export const navigationModulesFactory = (t: TFunction) => {
    const {
        PREGNANCY_MODULE_PARENT_NAV,
        NBC_AND_PNC_PARENT_NAV,
        NUTRITION_PARENT_NAVIGATION,
        CLIENT_MODULE_PARENT_NAV,
        HOME_PARENT_NAV,
        ADMIN_MODULE_PARENT_NAV,
        REPORT_MODULE_PARENT_NAV,
    } = modulePageLinks(t);
    const {
        PREGNANCY_COMPARTMENTS_PAGE_NAVIGATION,
        PREGNANCY_ANALYSIS_PAGE_NAVIGATION,
        PREGNANCY_LOGFACE_PAGE_NAVIGATION,
        NBC_AND_PNC_LOGFACE_PAGE_NAVIGATION,
        NBC_AND_PNC_COMPARTMENTS_PAGE_NAVIGATION,
        NBC_AND_PNC_ANALYSIS_PAGE_NAVIGATION,
        NUTRITION_LOGFACE_PAGE_NAVIGATION,
        NUTRITION_COMPARTMENTS_PAGE_NAVIGATION,
        NUTRITION_ANALYSIS_PAGE_NAVIGATION,
        CLIENT_PAGE_NAVIGATION,
        HOUSEHOLD_PAGE_NAVIGATION,
        ANC_PAGE_NAVIGATION,
        CHILD_PAGE_NAVIGATION,
        USERS_PAGE_NAVIGATION,
        ROLES_PAGE_NAVIGATION,
        TEAMS_PAGE_NAVIGATION,
        LOCATIONS_PAGE_NAVIGATION,
    } = navigationPageLinks(t);
    // nav module objects
    const navigationModules: Dictionary<NavigationModule> = {
        HOME_NAVIGATION_MODULE: {
            childNavs: [],
            parentNav: HOME_PARENT_NAV,
        },

        PREGNANCY_NAVIGATION_MODULE: {
            childNavs: [
                PREGNANCY_LOGFACE_PAGE_NAVIGATION,
                PREGNANCY_COMPARTMENTS_PAGE_NAVIGATION,
                PREGNANCY_ANALYSIS_PAGE_NAVIGATION,
            ].filter((childNav): childNav is PageLink => typeof childNav !== 'boolean'),
            parentNav: PREGNANCY_MODULE_PARENT_NAV,
        },

        NBC_AND_PNC_NAVIGATION_MODULE: {
            childNavs: [
                NBC_AND_PNC_LOGFACE_PAGE_NAVIGATION,
                NBC_AND_PNC_COMPARTMENTS_PAGE_NAVIGATION,
                NBC_AND_PNC_ANALYSIS_PAGE_NAVIGATION,
            ],
            parentNav: NBC_AND_PNC_PARENT_NAV,
        },

        NUTRITION_MODULE: {
            childNavs: [
                NUTRITION_LOGFACE_PAGE_NAVIGATION,
                NUTRITION_COMPARTMENTS_PAGE_NAVIGATION,
                NUTRITION_ANALYSIS_PAGE_NAVIGATION,
            ],
            parentNav: NUTRITION_PARENT_NAVIGATION,
        },

        CLIENT_NAVIGATION_MODULE: {
            childNavs: [
                ENABLE_CLIENT_PAGE && CLIENT_PAGE_NAVIGATION,
                ENABLE_HOUSEHOLD_PAGE && HOUSEHOLD_PAGE_NAVIGATION,
                ENABLE_ANC_PAGE && ANC_PAGE_NAVIGATION,
                ENABLE_CHILD_PAGE && CHILD_PAGE_NAVIGATION,
            ].filter((childNav): childNav is PageLink => typeof childNav !== 'boolean'),
            parentNav: CLIENT_MODULE_PARENT_NAV,
        },
        REPORT_NAVIGATION_MODULE: {
            childNavs: [],
            parentNav: REPORT_MODULE_PARENT_NAV,
        },
        ADMIN_NAVIGATION_MODULE: {
            childNavs: [
                ENABLE_USER_PAGE && USERS_PAGE_NAVIGATION,
                ENABLE_ROLE_PAGE && ROLES_PAGE_NAVIGATION,
                ENABLE_TEAM_PAGE && TEAMS_PAGE_NAVIGATION,
                ENABLE_LOCATION_PAGE && LOCATIONS_PAGE_NAVIGATION,
            ].filter<PageLink>((childNav: PageLink | boolean): childNav is PageLink => typeof childNav !== 'boolean'),
            parentNav: ADMIN_MODULE_PARENT_NAV,
        },
    };
    return navigationModules;
};

/** Enable Clients Module from its child pages */
export const ENABLE_CLIENT_RECORDS_MODULE =
    ENABLE_ANC_PAGE || ENABLE_CLIENT_PAGE || ENABLE_HOUSEHOLD_PAGE || ENABLE_HOUSEHOLD_PAGE;
export type ENABLE_CLIENT_RECORDS_MODULE = typeof ENABLE_CLIENT_RECORDS_MODULE;

/** Enable Admin Module from its child pages */
export const ENABLE_ADMIN_MODULE = ENABLE_USER_PAGE || ENABLE_ROLE_PAGE || ENABLE_TEAM_PAGE || ENABLE_LOCATION_PAGE;
export type ENABLE_ADMIN_MODULE = typeof ENABLE_ADMIN_MODULE;
