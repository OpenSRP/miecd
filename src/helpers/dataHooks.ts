import {
    OPENSRP_SECURITY_AUTHENTICATE,
    LOCATION_FILTER_PARAM,
    RISK_CATEGORY_FILTER_PARAM,
    SMS_TYPE_FILTER_PARAM,
    SEARCH_FILTER_PARAM,
    PREGNANCY_DETECTION,
    UPDATE_PREGNANCY_DETECTION,
    NUTRITION_REGISTRATION,
    UPDATE_NUTRITION_REGISTRATION,
    DEATH_REPORT,
    UPDATE_DEATH_REPORT,
} from '../constants';
import { getQueryParams } from 'containers/LogFace/utils';
import { TFunction } from 'i18next';
import { useQuery } from 'react-query';
import { useSelector } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { OpenSRPService } from 'services/opensrp';
import { getSmsDataByFilters } from 'store/ducks/sms_events';
import { TreeNode } from './locationHierarchy/types';
import { generateJurisdictionTree } from './locationHierarchy/utils';
import { toastToError, getRiskCatFilter } from './utils';
import { LogFaceModules } from 'configs/settings';

const selectSmsData = getSmsDataByFilters();

/** hook to get this users assignment details, hierarchy, team assigned and location the team is assigned
 * @param t - the translator function
 */
export const useUserAssignment = (t: TFunction) => {
    return useQuery(
        'userAssignment',
        () => {
            const opensrpService = new OpenSRPService(OPENSRP_SECURITY_AUTHENTICATE);
            return opensrpService.read('');
        },
        {
            select: (response) => {
                return {
                    userHierarchy: generateJurisdictionTree(response.locations),
                    user: response.user,
                    team: response.team.team,
                    location: response.team.team.location,
                };
            },
            onError: (err: Error) => toastToError(err?.message ?? t('An error occurred')),
        },
    );
};

/** redux hok to select filtered sms data based on provided filters
 * @param location - route location details, to get search params
 * @param module - the module
 * @param userLocationTree - for filtering based on the user's assigned level
 */
export const useGetLogFaceFilteredSms = (
    location: RouteComponentProps['location'],
    module: LogFaceModules,
    userHierarchy?: TreeNode,
) => {
    return useSelector((state) => {
        const userLocationIdFilter = getQueryParams(location)[LOCATION_FILTER_PARAM] as string;
        const riskCategoryFilter = getQueryParams(location)[RISK_CATEGORY_FILTER_PARAM] as string;
        const smsTypeFilter = getQueryParams(location)[SMS_TYPE_FILTER_PARAM] as string;
        const searchFilter = getQueryParams(location)[SEARCH_FILTER_PARAM] as string;
        const matchesId = (node: TreeNode) => node.model.id === userLocationIdFilter;
        const userLocationNode = userHierarchy?.first(matchesId);
        let smsFilter = smsTypeFilter ? [smsTypeFilter] : undefined;
        switch (smsTypeFilter) {
            case PREGNANCY_DETECTION:
                smsFilter = [PREGNANCY_DETECTION, UPDATE_PREGNANCY_DETECTION];
                break;
            case NUTRITION_REGISTRATION:
                smsFilter = [NUTRITION_REGISTRATION, UPDATE_NUTRITION_REGISTRATION];
                break;
            case DEATH_REPORT:
                smsFilter = [DEATH_REPORT, UPDATE_DEATH_REPORT];
                break;
            default:
                break;
        }
        return selectSmsData(state, {
            locationNode: userLocationNode,
            riskCategory: getRiskCatFilter(module, riskCategoryFilter),
            smsTypes: smsFilter,
            searchFilter,
            module,
        });
    });
};
