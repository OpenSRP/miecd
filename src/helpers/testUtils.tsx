import { mount, MountRendererProps } from 'enzyme';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { act } from 'react-dom/test-utils';

export const mountWithTranslations: typeof mount = (component: React.ReactNode, options?: MountRendererProps) => {
    return mount(<I18nextProvider i18n={i18n}>{component}</I18nextProvider>, options);
};

/**
 * Utility to await promises to resolve in tests
 * @param ms time in ms
 */
export async function waitForPromises() {
    await act(async () => {
        await new Promise((res) => setImmediate(res));
    });
}
