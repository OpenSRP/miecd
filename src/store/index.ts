import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Dictionary } from '@onaio/utils';

/** Initial reducers in the reducer registry */
const defaultReducers: Dictionary = {
    router: connectReducer /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
    store.replaceReducer(combine(reducers));
});

export default store;
