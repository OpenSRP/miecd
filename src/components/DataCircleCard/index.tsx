import React from 'react';
import { connect } from 'react-redux';
import { Link, RouteComponentProps, withRouter } from 'react-router-dom';
import { Card, CardBody, CardTitle } from 'reactstrap';
import './index.css';
import {
    RED,
    PREGNANCY,
    NBC_AND_PNC_CHILD,
    NBC_AND_PNC_WOMAN,
    NUTRITION,
    ALL,
    INAPPROPRIATELY_FED,
    NO,
    OVERWEIGHT,
    RISK,
    SEVERE_WASTING,
    STUNTED,
} from '../../constants';
import { getLinkToHierarchicalDataTable, Dictionary } from '../../helpers/utils';
import { addFilterArgs } from '../../store/ducks/sms_events';
import { SmsFilterFunction } from '../../types';
import './index.css';
import { useTranslation } from 'react-i18next';

/**
 * interface for props to be passed to DataCircleCard component.
 */
interface Props extends RouteComponentProps {
    redAlert?: number;
    risk?: number;
    noRisk?: number;
    totalChildren?: number;
    stunting?: number;
    wasting?: number;
    overweight?: number;
    inappropriateFeeding?: number;
    title: string;
    addFilterArgsActionCreator?: typeof addFilterArgs;
    filterArgs?: SmsFilterFunction[];
    module: PREGNANCY | NBC_AND_PNC_CHILD | NBC_AND_PNC_WOMAN | NUTRITION | '';
    className?: string;
    userLocationId: string;
    permissionLevel: number;
}

/**
 * functional component that takes in props
 */
interface CircleSpecProps {
    class: string;
    riskLabel: string;
    riskType: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    riskValue: any;
}

function DataCircleCard({
    redAlert,
    risk,
    noRisk,
    stunting,
    wasting,
    overweight,
    inappropriateFeeding,
    title,
    addFilterArgsActionCreator = addFilterArgs,
    filterArgs,
    module,
    className = '',
    userLocationId,
    permissionLevel,
}: Props) {
    const { t } = useTranslation();
    const pregnancyAndPncCircleSpec: CircleSpecProps[] = [
        {
            class: 'red',
            riskLabel: t('Red Alert'),
            riskType: RED,
            riskValue: redAlert,
        },
        {
            class: 'yellow',
            riskLabel: t('risk'),
            riskType: RISK,
            riskValue: risk,
        },
        {
            class: 'green',
            riskLabel: t('No Risk'),
            riskType: NO,
            riskValue: noRisk,
        },
    ];

    const nutritionCircleSpec: CircleSpecProps[] = [
        {
            class: 'stunting',
            riskLabel: t('Stunting'),
            riskType: STUNTED,
            riskValue: stunting,
        },
        {
            class: 'wasting',
            riskLabel: t('Wasting'),
            riskType: SEVERE_WASTING,
            riskValue: wasting,
        },
        {
            class: 'overweight',
            riskLabel: t('Overweight'),
            riskType: OVERWEIGHT,
            riskValue: overweight,
        },
        {
            class: 'inappropriate-feeding',
            riskLabel: t('Inappropriate Feeding'),
            riskType: INAPPROPRIATELY_FED,
            riskValue: inappropriateFeeding,
        },
    ];

    return (
        <Card className={`dataCircleCard ${className}`}>
            <CardTitle>
                <Link
                    to={getLinkToHierarchicalDataTable(ALL, module, title, permissionLevel, userLocationId)}
                    // tslint:disable-next-line: jsx-no-lambda
                    onClick={() => {
                        if (filterArgs) {
                            addFilterArgsActionCreator(filterArgs);
                        }
                    }}
                >
                    {title}
                </Link>
            </CardTitle>
            <CardBody className="circlesRow">
                {(module !== NUTRITION ? pregnancyAndPncCircleSpec : nutritionCircleSpec).map(
                    (spec: Dictionary, i: number) => (
                        <Link
                            className={`${spec.class} circle`}
                            key={i}
                            // tslint:disable-next-line: jsx-no-lambda
                            onClick={() => {
                                if (filterArgs) {
                                    addFilterArgsActionCreator(filterArgs);
                                }
                            }}
                            to={getLinkToHierarchicalDataTable(
                                spec.riskType,
                                module,
                                title,
                                permissionLevel,
                                userLocationId,
                            )}
                        >
                            <span className="number">{spec.riskValue}</span>
                            <span className="risk-level">{spec.riskLabel}</span>
                        </Link>
                    ),
                )}
            </CardBody>
        </Card>
    );
}

const mapDispatchToProps = { addFilterArgsActionCreator: addFilterArgs };

const ConnectedDataCircleCard = connect(null, mapDispatchToProps)(withRouter(DataCircleCard));

export default ConnectedDataCircleCard;
