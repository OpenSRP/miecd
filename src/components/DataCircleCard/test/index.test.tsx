import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { removeFilterArgs } from 'store/ducks/sms_events';
import DataCircleCard from '..';
import { PREGNANCY } from '../../../constants';
import { mountWithTranslations } from '../../../helpers/testUtils';
import store from '../../../store/index';
import * as smsEventsDucks from '../../../store/ducks/sms_events';

const history = createBrowserHistory();

describe('DataCircleCard', () => {
    afterEach(() => {
        store.dispatch(removeFilterArgs());
        jest.resetAllMocks();
    });
    it('must render correctly', () => {
        const props = {
            highRisk: 10,
            lowRisk: 10,
            noRisk: 10,
            title: 'Total Pregnancies',
            userLocationId: 'd1865325-11e6-4e39-817b-e676c1affecf',
            permissionLevel: 0,
            totalNumber: 2,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <DataCircleCard {...props} module={PREGNANCY} />
                </Router>
            </Provider>,
        );
        expect(toJson(wrapper.find('Card'))).toMatchSnapshot();
    });

    it('card title link works', () => {
        const filterArgsSpy = jest.fn();
        const props = {
            filterArgs: [filterArgsSpy],
            highRisk: 10,
            lowRisk: 10,
            noRisk: 10,
            title: 'Total Pregnancies',
            userLocationId: 'd1865325-11e6-4e39-817b-e676c1affecf',
            permissionLevel: 0,
            totalNumber: 2,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <DataCircleCard {...props} module={PREGNANCY} />
                </Router>
            </Provider>,
        );

        const cardTitleLink = wrapper.find('CardTitle Link');
        expect(toJson(cardTitleLink)).toMatchSnapshot('card title link');

        // simulate click on card title link
        cardTitleLink.simulate('click', { button: 0 });
        wrapper.update();
        expect(smsEventsDucks.getFilterArgs(store.getState())).toEqual([filterArgsSpy]);
        wrapper.unmount();
    });

    it('card body link works', () => {
        const filterArgsSpy = jest.fn();
        const props = {
            filterArgs: [filterArgsSpy],
            highRisk: 10,
            lowRisk: 10,
            noRisk: 10,
            title: 'Total Pregnancies',
            userLocationId: 'd1865325-11e6-4e39-817b-e676c1affecf',
            permissionLevel: 0,
            totalNumber: 2,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <DataCircleCard {...props} module={PREGNANCY} />
                </Router>
            </Provider>,
        );

        const cardBodyLink = wrapper.find('CardBody Link').first();
        expect(toJson(cardBodyLink)).toMatchSnapshot('card title link');

        // simulate click on card title link
        cardBodyLink.simulate('click', { button: 0 });
        wrapper.update();
        expect(smsEventsDucks.getFilterArgs(store.getState())).toEqual([filterArgsSpy]);
        wrapper.unmount();
    });

    it('card header shows total', () => {
        const filterArgsSpy = jest.fn();
        const props = {
            filterArgs: [filterArgsSpy],
            highRisk: 10,
            lowRisk: 10,
            noRisk: 10,
            title: 'Total Pregnancies',
            userLocationId: 'd1865325-11e6-4e39-817b-e676c1affecf',
            permissionLevel: 0,
            totalNumber: 14,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <DataCircleCard {...props} module={PREGNANCY} />
                </Router>
            </Provider>,
        );

        // expect card title to match snapshot
        const cardTitle = wrapper.find('CardTitle .card_title').text();
        expect(cardTitle).toBe('14 Total Pregnancies');

        wrapper.unmount();
    });
});
