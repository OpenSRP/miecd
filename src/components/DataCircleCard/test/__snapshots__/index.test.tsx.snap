// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataCircleCard card body link works: card title link 1`] = `
<Link
  className="red circle"
  key="0"
  onClick={[Function]}
  to="/pregnancy_compartments/hierarchicaldata/Pregnancy/red/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
>
  <LinkAnchor
    className="red circle"
    href="/pregnancy_compartments/hierarchicaldata/Pregnancy/red/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
    navigate={[Function]}
    onClick={[Function]}
  >
    <a
      className="red circle"
      href="/pregnancy_compartments/hierarchicaldata/Pregnancy/red/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
      onClick={[Function]}
    >
      <span
        className="number"
      />
      <span
        className="risk-level"
      >
        Red Alert
      </span>
    </a>
  </LinkAnchor>
</Link>
`;

exports[`DataCircleCard card title link works: card title link 1`] = `
<Link
  onClick={[Function]}
  to="/pregnancy_compartments/hierarchicaldata/Pregnancy/all/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
>
  <LinkAnchor
    href="/pregnancy_compartments/hierarchicaldata/Pregnancy/all/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
    navigate={[Function]}
    onClick={[Function]}
  >
    <a
      href="/pregnancy_compartments/hierarchicaldata/Pregnancy/all/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
      onClick={[Function]}
    >
      <h5
        className="card_title"
      >
        2 Total Pregnancies
      </h5>
    </a>
  </LinkAnchor>
</Link>
`;

exports[`DataCircleCard must render correctly 1`] = `
<Card
  className="dataCircleCard"
  tag="div"
>
  <div
    className="dataCircleCard card"
  >
    <CardTitle
      tag="div"
    >
      <div
        className="card-title"
      >
        <Link
          onClick={[Function]}
          to="/pregnancy_compartments/hierarchicaldata/Pregnancy/all/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
        >
          <LinkAnchor
            href="/pregnancy_compartments/hierarchicaldata/Pregnancy/all/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              href="/pregnancy_compartments/hierarchicaldata/Pregnancy/all/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
              onClick={[Function]}
            >
              <h5
                className="card_title"
              >
                2 Total Pregnancies
              </h5>
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </CardTitle>
    <CardBody
      className="circlesRow"
      tag="div"
    >
      <div
        className="circlesRow card-body"
      >
        <Link
          className="red circle"
          key="0"
          onClick={[Function]}
          to="/pregnancy_compartments/hierarchicaldata/Pregnancy/red/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
        >
          <LinkAnchor
            className="red circle"
            href="/pregnancy_compartments/hierarchicaldata/Pregnancy/red/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              className="red circle"
              href="/pregnancy_compartments/hierarchicaldata/Pregnancy/red/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
              onClick={[Function]}
            >
              <span
                className="number"
              />
              <span
                className="risk-level"
              >
                Red Alert
              </span>
            </a>
          </LinkAnchor>
        </Link>
        <Link
          className="yellow circle"
          key="1"
          onClick={[Function]}
          to="/pregnancy_compartments/hierarchicaldata/Pregnancy/risk/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
        >
          <LinkAnchor
            className="yellow circle"
            href="/pregnancy_compartments/hierarchicaldata/Pregnancy/risk/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              className="yellow circle"
              href="/pregnancy_compartments/hierarchicaldata/Pregnancy/risk/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
              onClick={[Function]}
            >
              <span
                className="number"
              />
              <span
                className="risk-level"
              >
                risk
              </span>
            </a>
          </LinkAnchor>
        </Link>
        <Link
          className="green circle"
          key="2"
          onClick={[Function]}
          to="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
        >
          <LinkAnchor
            className="green circle"
            href="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              className="green circle"
              href="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/2 Total Pregnancies/0/down/d1865325-11e6-4e39-817b-e676c1affecf/0"
              onClick={[Function]}
            >
              <span
                className="number"
              >
                10
              </span>
              <span
                className="risk-level"
              >
                No Risk
              </span>
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </CardBody>
  </div>
</Card>
`;
